app/src/main/java/space/taran/arknavigator/ui/fragments/preview/PreviewAndThumbnail.kt

```diff
@@ -0,0 +1,149 @@
+package space.taran.arknavigator.ui.fragments.preview
+
+import android.graphics.Bitmap
+import android.graphics.drawable.Drawable
+import android.util.Log
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.RequestOptions
+import com.bumptech.glide.request.target.CustomTarget
+import com.bumptech.glide.request.transition.Transition
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import space.taran.arknavigator.mvp.model.repo.index.ResourceId
+import space.taran.arknavigator.mvp.model.repo.index.ResourceMeta
+import space.taran.arknavigator.mvp.model.repo.extra.ImageMetaExtra
+import space.taran.arknavigator.ui.App
+import space.taran.arknavigator.utils.PREVIEWS
+import space.taran.arknavigator.utils.extension
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+import kotlin.io.path.notExists
+
+data class PreviewAndThumbnail(val preview: Path, val thumbnail: Path) {
+
+    //todo: separate PreviewRepos similar to ResourceIndexes
+    // this would replicate generated previews to other devices
+    companion object {
+
+        private val PREVIEWS_STORAGE: Path =
+            Paths.get("${App.instance.cacheDir}/previews")
+        private val THUMBNAILS_STORAGE: Path =
+            Paths.get("${App.instance.cacheDir}/thumbnails")
+
+        private const val THUMBNAIL_WIDTH = 72
+        private const val THUMBNAIL_HEIGHT = 128
+        private const val COMPRESSION_QUALITY = 100
+
+        private fun previewPath(id: ResourceId): Path =
+            PREVIEWS_STORAGE.resolve(id.toString())
+        private fun thumbnailPath(id: ResourceId): Path =
+            THUMBNAILS_STORAGE.resolve(id.toString())
+
+        init {
+            if (PREVIEWS_STORAGE.notExists()) Files.createDirectories(PREVIEWS_STORAGE)
```

Comment:
Actually `Files.createDirectories` creates the directory if it doesn't exist.
That distinguishes this method from `Files.createDirectory` if I remember correctly.